# Generated by Django 3.0.4 on 2020-05-18 21:06

from django.db import migrations

# sample number in PowderSample is erroneously reich_lab_id instead of sample database id
def fix_sample_id(apps, schema_editor):
	PowderSample = apps.get_model('samples', 'PowderSample')
	Sample = apps.get_model('samples', 'Sample')
	
	for powder_sample in PowderSample.objects.all():
		sample_id_db = powder_sample.sample.id
		corrected_sample = Sample.objects.get(id=sample_id_db)
		
		if str(corrected_sample.reich_lab_id) not in powder_sample.powder_sample_id:
			raise ValueError('Error for powder sample: ' + str(powder_sample) + ' ' + str(corrected_sample))
		
		powder_sample.sample = corrected_sample
		powder_sample.save()
		
def reverse_sample_id(apps, schema_editor):
	PowderSample = apps.get_model('samples', 'PowderSample')
	Sample = apps.get_model('samples', 'Sample')
	
	for powder_sample in PowderSample.objects.all():
		sample_id_db = powder_sample.sample.id
		original_sample = Sample.objects.get(reich_lab_id=sample_id_db)
		powder_sample.sample = original_sample
		powder_sample.save()

class Migration(migrations.Migration):

    dependencies = [
        ('samples', '0005_results_library_foreign_key'),
    ]

    operations = [
		migrations.RunPython(fix_sample_id, reverse_sample_id),
    ]
